interface/pages/posts/new_sample.tsx

```diff
@@ -33,176 +39,165 @@ export default function NewSample() {
-        },
-        body: JSON.stringify(data),
-      }
-    )
-    const sample = await response.json()
+    const response = await state.postSample(data)
     if (response.status == 200) {
       console.log("POST /samples")
-      console.dir(sample)
     } else {
       console.log("POST /samples failed.")
     }
   }
 
-  const { classes } = useStyles();
-  const [value, setValue] = useState(null);
-  const theme = useMantineTheme();
-
   return (
-    <>
-      <AppShell
-        className={classes.app}
-        padding="md"
-        styles={(theme) => ({
-          main: { backgroundColor: theme.colorScheme === 'dark' ? theme.colors.dark[8] : theme.colors.gray[0] },
-        })}
-        header={MyHeader()}
-        footer={MyFooter()}
+    <AppShell
+      className={classes.app}
+      padding="md"
+      styles={(theme) => ({
+        main: { backgroundColor: theme.colorScheme === 'dark' ? theme.colors.dark[8] : theme.colors.gray[0] },
+      })}
+      header={MyHeader()}
+      footer={MyFooter()}
+    >
+      <Title order={2} mt="md">
+        Add a new sample
+      </Title>
+
+      <Divider mt="lg" />
+
+      <form
+        onSubmit={form.onSubmit(
+          async (values) => await postSample({
+            person_id: values.person_id,
+            location_id: values.location_id,
+            timestamp: values.timestamp,
+            sex: null,
+            lifestage: null,
+            reproduction: null,
+            image_url: values.image_url,
+            image_timestamp: values.image_timestamp,
+            image_desc: values.image_desc,
+          })
+        )}
       >
-        <Title order={2} mt="md">
-          Add a new sample
-        </Title>
-
-        <Divider mt="lg" />
-
-        <form
-          onSubmit={form.onSubmit(
-            async (values) => await postSample({
-              person_id: values.person_id,
-              location_id: values.location_id,
-              timestamp: values.timestamp,
-              image_url: values.image_url,
-              image_timestamp: values.image_timestamp,
-              image_desc: values.image_desc,
-            })
-          )}
-        >
-          <Stack spacing={20} mt="md">
-            <TextInput
-              placeholder="Person ID"
-              label="Person ID:"
-              sx={{ width: 100 }}
-              withAsterisk
-              {...form.getInputProps('person_id')}
-            />
-            <Group>
-              <Stack>
-                <label htmlFor="extractiontime">Extraction datatime:</label>
-                <Group>
-                  <input
-                    type="datetime-local"
-                    id="timestamp"
-                    name="timestamp"
-                    style={{ width: 200 }}
-                    {...form.getInputProps('timestamp')}
+        <Stack spacing={20} mt="md">
+          <Select
+            label="Person"
+            sx={{ width: 200 }}
+            data={state.persons.map(p => (
```

Comment:
Here is an example of using data of the db elsewhere than on the `persons` pages for ex. It does not cause any delay as the data is already in memory